tools/ci_build/github/azure-pipelines/nuget/templates/dml-vs-2022.yml

```diff
@@ -231,6 +231,15 @@ stages:
               searchPattern: '**/*.pdb'
               symbolServerType: teamServices
 
+      - ${{ if eq(parameters['DoCompliance'], 'true') }}:
+        - template: ../../templates/compliance.yml
```

Comment:
Some compliance tasks, for example Prefast, will modify the binaries we publish. That's we should publish the artifacts first. 
If the component-governance is the main concern, we can set the task's "continueOnError" to true.

Reply:
I have removed all Prefast tasks.  Now we use CodeQL instead. So, the concern regarding to Prefast doesn't hold anymore.  I don't know if any other compliace task might also have this issue.  It's ok to have this change. I prefer to adding "continueOnError" instead.

Reply:
The pipeline A status will not be passed if any stage's status isn't green.
As a result, the pipeline B which depends on A will not be triggered.